plugins/attachment-resources/src/utils.ts

```diff
@@ -84,3 +86,19 @@ export function getType (type: string): 'image' | 'video' | 'audio' | 'pdf' | 'o
 
   return 'other'
 }
+
+export async function moveAttachments (doc: Doc, space: Ref<Space>): Promise<void> {
+  const client = getClient()
+  const hierarchy = client.getHierarchy()
+  const attributes = hierarchy.getAllAttributes(doc._class)
+  for (const [name, attribute] of attributes) {
+    if (hierarchy.isDerived(attribute.type._class, core.class.Collection)) {
+      const collection = attribute.type as Collection<AttachedDoc>
+      const allAttached = await client.findAll(collection.of, { attachedTo: doc._id })
+      for (const attached of allAttached) {
+        moveAttachments(attached, space).catch((err) => console.log('failed to move', name, err))
```

Comment:
It is bad way to mix async and promise concepts, please use async approach during one function code

Reply:
Fixed